lib/govuk_navigation_helpers.rb

```diff
@@ -1,5 +1,39 @@
 require "govuk_navigation_helpers/version"
 
 module GovukNavigationHelpers
-  # Your code goes here...
+  class NavigationHelper
+    def initialize(content_item)
+      @content_item = content_item
+    end
+
+    # Generate a breacrumb trail
+    #
+    # @return [Array<Hash>] Each item is a hash containing `:title` and `:url` for one link in the breadcrumb
+    def breadcrumbs
+      root = {title: "Home", url: "/"}
+      direct_parent = parent
+      ordered_parents = []
+
+      while direct_parent
+        ordered_parents.unshift(
+          title: direct_parent["title"],
+          url: direct_parent["base_path"],
+        )
+        direct_parent = direct_parent["parent"] && direct_parent["parent"].first
+      end
+
+      ordered_parents.unshift(root)
+    end
+
+  private
+    attr_reader :content_item
+
+    def links
+      @links ||= content_item["links"]
+    end
+
+    def parent
+      links["parent"].first if links.include?("parent")
```

Comment:
I took out `dig` because it was introduced quite recently. The version of ruby used by government-frontend didn't have it.